src/distributed/distribute_optigraph.jl

```
@@ -0,0 +1,161 @@
+end
+
+function _convert_local_to_remote(func::GenericNonlinearExpr{Plasmo.NodeVariableRef}, variable_map::Dict{NodeVariableRef, RemoteVariableRef})
+    V = Plasmo.RemoteVariableRef
+    ret = JuMP.GenericNonlinearExpr{V}(func.head, Any[])
+    stack = Tuple{JuMP.GenericNonlinearExpr, Any}[]
+
+    for arg in reverse(func.args)
+        push!(stack, (ret, arg))
+    end
+    while !isempty(stack)
+        parent, arg = pop!(stack)
+        if arg isa GenericNonlinearExpr
+            new_ret = JuMP.GenericNonlinearExpr{V}(arg.head, Any[])
+            push!(parent.args, new_ret)
+            for child in reverse(arg.args)
+                push!(stack, (new_ret, _convert_local_to_remote(child, variable_map)))
+            end
+        else
+            push!(parent.args, _convert_local_to_remote(arg, variable_map))
+        end
+    end
+    return ret
+end
+
+function _convert_local_to_remote(func::NodeVariableRef, variable_map::Dict{NodeVariableRef, RemoteVariableRef})
+    return variable_map[func]
+end
+
+function _convert_local_to_remote(func::RemoteVariableRef, variable_map::Dict{NodeVariableRef, RemoteVariableRef})
+    return func
+end
+
+function _convert_local_to_remote(func::Float64, variable_map::Dict{NodeVariableRef, RemoteVariableRef})
+    return func
+end
+
+function _convert_local_to_remote(func::Array{Float64}, variable_map::Dict{NodeVariableRef, RemoteVariableRef})
+    return func
+end
+
+function _convert_local_to_remote(func::GenericAffExpr{Float64, RemoteVariableRef}, variable_map::Dict{NodeVariableRef, RemoteVariableRef})
+    return func
+end
+
+function _convert_local_to_remote(func::GenericQuadExpr{Float64, RemoteVariableRef}, variable_map::Dict{NodeVariableRef, RemoteVariableRef})
+    return func
+end
+
+function _convert_local_to_remote(func::GenericNonlinearExpr{RemoteVariableRef}, variable_map::Dict{NodeVariableRef, RemoteVariableRef})
+    return func
+end
```

there is probably some generic fallback type we can use here. maybe even `Any`?

---

I think that would the code more clean. My main concern is that I think if it ends up with a different type, it should maybe throw an error? These functions are defined mainly for [this line](https://github.com/dlcole3/Plasmo.jl/blob/f801ebb4519c4b82c6c52448de5b70df18a28869/src/distributed/distribute_optigraph.jl#L87) where we are building the nonlinear expression. This `_convert_local_to_remote` can get called on lots of things just in the NLP converter, so I don't use `Any` here in case there is some edge case I did not know about so that it would throw an error. 

As an example, in the `local_conversions.jl` and `remote_conversions.jl` files, I add a separate catch with `Any` that will throw a warning [here](https://github.com/dlcole3/Plasmo.jl/blob/f801ebb4519c4b82c6c52448de5b70df18a28869/src/distributed/local_conversions.jl#L445-L465) in the event that something is getting passed and being serialized/deserialized unexpectedly. Do you think we should still just make these `Any`? For the `local_conversions.jl` and `remote_conversions.jl` files, I am inclined to keep the warning, but for the `distirbute_optigraph.jl` file, we could maybe switch to a single `Any` function

---

`Any` is probably too unsafe of a fallback to be honest. I think what you have is perfectly fine for now. We could probably define a union over some of the numeric types and JuMP-specific types. I'll try a few ideas.